docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+        descriptions containing the user's input.
+
+`NoteList` is also an `ArrayList` of type `Displayable`, which is one of two interfaces implemented in the code for 
+E-Duke-8. As such, many of the commands that manipulate the `TopicList` make use of the package `java.util.ArrayList`. 
+The `NoteList` stores `Note` objects. Each topic has 1 `NoteList`. 
+
+#### 2.2.3. Implementation of Notes
+
+**Adding a new note:**
+
+This task is performed by the `NoteList.add()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.add()` method. 
+        A new `Note` object is passed into its parameter.
+
+Step 2: The `NoteList.add()` method makes use of `ArrayList` API, specifically the `ArrayList.add()` method, to add 
+        the `Note` object into `NoteList`.
+
+**Deleting a note:**
+
+This task is performed by the `NoteList.add()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then calls the `NoteList.delete()` method. 
+        An integer that represents the index of the `Note` object to be deleted within the `NoteList` is passed into 
+        this method.
+
+Step 2: The `NoteList.add()` method makes use of `ArrayList` API, specifically the `ArrayList.remove()` method, to 
+        delete the `Note` object in `NoteList`.
+
+**Listing out all notes in a topic**
+
+This task is performed by the `Topic.showNotes()` method.
+
+Step 1: The `parseCommand()` method instantiates a `TopicCommand` object which then calls the `Topic.showNotes()` method. 
+
+Step 2: The `Topic.showNotes()` method calls the `Ui.printNoteList()` method. The topic's `NoteList` into this method. 
+        `Ui.printNoteList()` prints out all the descriptions of the `Note` objects in the `NoteList`.
+
+#### 2.2.4. Design of Option and OptionList 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
+
+### 2.3. Logic Component
+
+#### 2.3.1. Design of Parser
+
+![Parser Diagram](./images/ParserDiagram.png)
+
+1. After constructing a new MenuParser() in the `Eduke8` class, the parseCommand() method is used to parse the 
+   user command.
+2. This results in a `Command` object, which is executed by `Command` class itself, using the execute() method.
+3. The `Ui` object in the `Command` object is used to display the requested information, or to display the required task 
+to be completed as per the user input.
+
+#### 2.3.2. Implementation of MenuParser
+
+Below is the sequence diagram for how the Parser component of `Eduke8` works with commands to show output to the user.
+
+![Parser Sample Sequence](./images/ParserSampleSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/56961189/97391586-810eb180-191a-11eb-8a82-5431b39ec4f4.png)

1. Should :MenuParser be deactivated right at "return new HelpCommand()"?
2. Perhaps you can consider not splitting :Eduk8's activation bar, that is to say, a full activation bar from "ui.getInputFromUser()" to the end. Should there also be a return dotted line pointing from :Eduk8 at the end of the activation bar?
3. Should there be a return dotted line from :HelpCommand to :Eduk8 after command.execute()?
4. Should :Ui return be pointing to an activation bar?

Reply:
1. Is there a missing "e" in Eduke8?
2. Should the return new HelpCommand() dotted line start from the bottom of the :MenuParser activation bar instead?